eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/ReplicationEndpoint.scala

```diff
@@ -363,6 +359,96 @@ class ReplicationEndpoint(
 }
 
 /**
+ * An [[EndpointFilter]] computes a [[ReplicationFilter]] that shall be applied to a
+ * replication read request that replicates from a source log (defined by ``sourceLogName``)
+ * to a target log (defined by ``targetLogId``).
+ */
+trait EndpointFilter {
+  def filterFor(targetLogId: String, sourceLogName: String): ReplicationFilter
+}
+
+object EndpointFilter {
+  private class CombiningEndpointFilter(
+    targetFilters: Map[String, ReplicationFilter],
+    sourceFilters: Map[String, ReplicationFilter],
+    targetSourceCombinator: (ReplicationFilter, ReplicationFilter) => ReplicationFilter) extends EndpointFilter {
+
+    override def filterFor(targetLogId: String, sourceLogName: String): ReplicationFilter = {
+      (targetFilters.get(targetLogId), sourceFilters.get(sourceLogName)) match {
+        case (None, Some(sourceFilter))               => sourceFilter
+        case (Some(targetFilter), None)               => targetFilter
+        case (None, None)                             => NoFilter
+        case (Some(targetFilter), Some(sourceFilter)) => targetSourceCombinator(targetFilter, sourceFilter)
+      }
+    }
+  }
+
+  /**
+   * Creates an [[EndpointFilter]] that computes a [[ReplicationFilter]] for a replication read request
+   * from a source log to a target log by and-combining target and source filters when given in the provided [[Map]]s.
+   * If only source or target filter is given that is returned. If no filter is given [[com.rbmhtechnology.eventuate.ReplicationFilter.NoFilter]] is returned.
+   *
+   * This [[EndpointFilter]] can be useful when an eventsourced service that is not replicated collaborates
+   * with different external service. In that case `sourceFilters` can ensure that no ''internal'' event is
+   * replicated to other services and `targetFilters` can add target specific filter criteria.
```

Comment:
See also https://github.com/RBMHTechnology/eventuate/pull/303/commits/05f58abd6c0d37d8fe44e97277989355e0a8fb80#r77624524 regarding the term *service*